bolt-modules/boltlib/lib/puppet/functions/get_resources.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'bolt/task'
+
+# Query the state of resources on a list of targets using resource definitions in Bolt's modulepath.
+# The results are returned as a list of hashes representing each resource.
+#
+# Requires the Puppet Agent be installed on the target, which can be accomplished with apply_prep
+# or by directly running the puppet_agent::install task.
+Puppet::Functions.create_function(:get_resources) do
+  # @param targets A pattern or array of patterns identifying a set of targets.
+  # @param resources A resource type or instance, or an array of such.
+  # @example Collect resource states for packages and a file
+  #   get_resources('target1,target2', [Package, File[/etc/puppetlabs]])
+  dispatch :get_resources do
+    param 'Boltlib::TargetSpec', :targets
+    param 'Variant[String, Resource, Array[Variant[String, Resource]]]', :resources
```

Comment:
We should probably have a regex for the strings. perhaps inside the function so we can provide a better error.